lib/StaticAnalyzer/Core/PlistDiagnostics.cpp

```diff
@@ -917,7 +920,9 @@ static MacroNameAndArgs getMacroNameAndArgs(SourceLocation ExpanLoc,
   assert(II && "Failed to acquire the IndetifierInfo for the macro!");
 
   const MacroInfo *MI = getMacroInfoForLocation(PP, SM, II, ExpanLoc);
-  assert(MI && "The macro must've been defined at it's expansion location!");
+  // assert(MI && "The macro must've been defined at it's expansion location!");
```

Comment:
Could you please add a comment instead of this line detailing that with the use of CTU we may not be able to acquire the `MacroInfo` object? Without it, the assert should stay, but making an if branch for this case (and threading `AnalyzerOptions` through all of these functions) is definitely not worth the chore.

Reply:
How about this:

We should always be able to obtain the `MacroInfo` in a given TU, but if we're running the analyzer with CTU, the Preprocessor won't contain the directive history (or anything for that matter) from another TU.
TODO: assert when we're not running with CTU